### PR TITLE
Bepis design: Table with wheels part 2: electric boogaloo

### PR DESCRIPTION
**Pull request by ghost**
🆑 Qustinnus & Kryson
add: new BEPIS design: a table with WHEELS
/🆑

hello gamers. Have you ever wanted felt like tables lacked something? Worry no more.

this time I found a better solution.

### CODE REVIEW

**Comment by Fikou** [comment]
bro you have this shit called like 4 different things 
is it a rolling table, a rolly poly, a table-with-wheels or an advanced wheel application

**Comment by Qustinnus** [comment]
yeah well ur fikou

**Comment by cacogen** [comment]
this is the only thing that has made me want to bother with that thing

**Comment by Fikou** [comment]
> yeah well ur fikou

reported to police

File: code/game/objects/structures/tables_racks.dm [hunk 1]
```diff
@@ -208,6 +214,36 @@
...
+/obj/structure/table/rolling/AfterPutItemOnTable(obj/item/I, mob/living/user)
+	. = ..()
+	attached_items += I
+	RegisterSignal(I, COMSIG_MOVABLE_MOVED, .proc/RemoveItemFromTable) //Listen for the pickup event, unregister on pick-up so we aren't moved
+
+/obj/structure/table/rolling/proc/RemoveItemFromTable(datum/source, newloc, dir)
+	if(newloc != loc) //Did we not move with the table? because that shit's ok
+		return FALSE
+	attached_items -= source
+	UnregisterSignal(source, COMSIG_ITEM_PICKUP)
```
Comment (nemvar) [inline code/game/objects/structures/tables_racks.dm]: wrong singal

**Comment by trollbreeder** [comment]
Socialists estimate in 1950, 83% of the population identified with some sort of religion.
That's more than 2 billion god-fearing people.

Do you think any of them stopped to wonder if we have become the gods?

**Comment by cacogen** [comment]
why is it a giant tank sprited in the style of risk of rain 1
why do we have to make things for it for it to serve a purpose
why do i have scientists begging me for money which i freely give them because unless i'm an assistant i have too much for it to be worth anything

the subtle genius of a table with wheels is not to be understated
a rolly poly is something an english child calls a forward roll which disappoints the idea

File: code/game/objects/structures/tables_racks.dm [hunk 2]
```diff
@@ -167,6 +167,9 @@
 	if(istype(I, /obj/item/storage/bag/tray))
 		var/obj/item/storage/bag/tray/T = I
 		if(T.contents.len > 0) // If the tray isn't empty
+			for(var/x in T.contents)
+				var/obj/item/item = x
+				AfterPutItemOnTable(item, user)
```
Comment (optimumtact) [inline code/game/objects/structures/tables_racks.dm]: this should have been a signal

Comment (Qustinnus) [reply code/game/objects/structures/tables_racks.dm]: I'll keep that in mind for a follow-up PR, thank you.

**Pull request merged**